frontend/src/metabase/containers/ItemPicker.jsx

```diff
@@ -76,6 +76,32 @@ export default class ItemPicker extends React.Component {
     }
   }
 
+  _checkHasWritePermissionForItem(item, models) {
+    const { collectionsById } = this.props;
+
+    // if user is selecting a collection, they must have a `write` access to it
+    if (models.has("collection") && item.model === "collection") {
+      return item.can_write;
+    }
+
+    // if user is selecting something else (e.g. dashboard),
+    // they must have `write` access to a collection item belongs to
+    const collection = item.collection_id
+      ? collectionsById[item.collection_id]
+      : collectionsById["root"];
+    return collection.can_write;
+  }
+
+  _checkCollectionHasChildWithWritePermission(collection) {
+    const hasChildren = collection.children.length > 0;
+    if (!hasChildren) {
+      return collection.can_write;
+    }
+    return collection.children.some(child =>
+      this._checkCollectionHasChildWithWritePermission(child),
+    );
```

Comment:
What about collections that are themselves writable, have children, but none of the children are writable? It seems like the logic should be `this is writable || this has writable children`. I haven't tested it though so I might be missing something.

Reply:
You're right, they get hidden in this case. Good catch, thank you!